migration/idempotent/008-tracing-functions.sql

```diff
@@ -355,7 +355,7 @@ BEGIN
             _value
         FROM _ps_trace.tag_key k
         WHERE k.key = _key
-        ON CONFLICT (key, value) DO
+        ON CONFLICT (key, pg_catalog.md5(value::TEXT)) DO
```

Comment:
I think at the end of this function we should check if there was a collision (IF _tag.value != _value) and throw an error if that's the case. That will verify that in even in the highly-unlikely case of a hash collision, we don't get data corruption. Instead we get an error.

Reply:
Good call. Do you think it's enough to guard against collisions during ingestion only? Or should I apply similar logic on reads? 

E.g.:
```
AND pg_catalog.md5(x.value::TEXT) = pg_catalog.md5(a.value::TEXT)
AND x.value = a.value
```

A potential downside is a heap lookup even in queries where `value` isn't used for anything else.